src/jsDelivr/index.js

```diff
@@ -0,0 +1,94 @@
+import got from 'got';
+import log from '../log.js';
+import config from '../config.js';
+import datadog from '../datadog.js';
+
+const hits = new Map();
+
+function formatHits(pkg) {
+  hits.set(pkg.name, pkg.hits);
+}
+
+/**
+ * Load downloads hits
+ */
+async function loadHits() {
+  const start = Date.now();
+  log.info('📦  Loading hits from jsDelivr');
+
+  try {
+    const { body: hitsJSON } = await got(config.jsDelivrHitsEndpoint, {
+      json: true,
+    });
+    hits.clear();
+    hitsJSON.forEach(formatHits);
+  } catch (e) {
+    log.error(e);
+  }
+
+  datadog.timing('jsdelivr.loadHits', Date.now() - start);
+}
+
+/**
+ * Get download hits
+ * @param {array} pkgs
+ */
+function getHits(pkgs) {
+  const start = Date.now();
+  const all = pkgs.map(({ name }) => {
+    const jsDelivrHits = hits.get(name) || 0;
+
+    return {
+      jsDelivrHits,
+      _searchInternal: {
+        // anything below 1000 hits/month is likely to mean that
+        // someone just made a few random requests so we count that as 0
+        jsDelivrPopularity: Math.max(jsDelivrHits.toString().length - 3, 0),
+      },
+    };
+  });
+
+  datadog.timing('jsdelivr.getHits', Date.now() - start);
+  return all;
+}
+
+/**
+ * Get packages files list
+ * @param {array} pkgs
+ */
+async function getAllFilesList(pkgs) {
+  const start = Date.now();
+
+  const files = await Promise.all(pkgs.map(getFilesList));
+
+  datadog.timing('jsdelivr.getAllFilesList', Date.now() - start);
+  return files;
+}
+
+/**
+ * Get one package files list
+ * @param {object} pkg
+ */
+async function getFilesList(pkg) {
+  const start = Date.now();
+  if (!pkg.name || !pkg.name.includes('@')) {
+    throw new Error(
+      `Package name should contain a version number: ${pkg.name}`
+    );
+  }
+
+  let files;
+  try {
+    files = await got(`${config.jsDelivrPackageEndpoint}/${pkg.name}/flat`, {
```

Comment:
we can put the return inside this try to avoid having a `let`. Not really a big deal though.

Annoying thing is that eslint might complain about "inconsistent return", but I'm not sure it will

Reply:
It was more to have a single metric call that way we don't forget to count a branch in the metric.